packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -6221,13 +6221,18 @@ class DefaultAssetBundle extends InheritedWidget {
 /// A given render object can be placed at most once in the widget tree. This
 /// widget enforces that restriction by keying itself using a [GlobalObjectKey]
 /// for the given render object.
+///
+/// This widget will call [RenderObject.dispose] on the [renderBox] when it is
```

Comment:
It's somewhat surprising that the adaptor takes ownership of the passed in `renderBox` and disposes it, no?

I originally expected that you'd be in charge of doing that in `onUnmount` if you don't want to reuse it?

Reply:
(Maybe also document this in the `renderBox` property?

Reply:
My thought is to keep this as a less-breaking change. Otherwise users will have to update their code to use the unmount property now where they didn't before. Updated the `renderBox` property.

Perhaps at some point it would be worth changing this, but my assumption is that most users use this widget as if it will dispose the render object, rather than caching a render object to use repeatedly with this widget.

Reply:
I see. I was thinking that maybe the default onUnmount handler would just dispose the renderBox, but if you don't want that you could provide a custom one and make it to do something else (e.g. keep the renderObject alive longer).

But maybe that can be added later. Maybe we should however specifically call out in the doc here that the provided renderBox becomes unusable when the adapter is unmounted from the tree?

Reply:
Added another line explaining that.